fix kubectl set command

WanLinghao opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:

1. fix some errors in 'kubectl set' series command

2.remove unused function in kubectl set env command

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```
NONE
```

WanLinghao commented:
@zjj2wry  PTAL

zjj2wry commented on pkg/kubectl/cmd/set/set_env.go:
```diff
@@ -74,10 +74,10 @@ var (
 
 	  # Import environment from a secret
-	  kubectl set env --from=secret/mysecret dc/myapp
+	  kubectl set env --from=secret/mysecret deployment/myapp
```
why update this?

WanLinghao replied:
kubectl can not recognize dc as abbreviation of  deployment config

WanLinghao replied:
@zjj2wry WDYT

zjj2wry replied:
yes, you are right

zjj2wry commented on pkg/kubectl/cmd/set/set_env.go:
```diff
@@ -123,8 +123,7 @@ type EnvOptions struct {
 
 	Cmd *cobra.Command
 
-	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
```
should update `f.UpdatePodSpecForObject()`to `o.UpdatePodSpecForObject()` better

WanLinghao replied:
OK

zjj2wry commented:
/assign

zjj2wry commented:
@WanLinghao thank you do this.

zjj2wry commented:
/Ok-to-test

zjj2wry commented:
/ok-to-test

BenTheElder commented:
/retest

WanLinghao committed:
fix some small errors in 'kubectl set' series command

modified:   pkg/kubectl/cmd/set/set_env.go
modified:   pkg/kubectl/cmd/set/set_serviceaccount.go
modified:   pkg/kubectl/cmd/set/set_subject.go

WanLinghao commented:
@fabianofranz @sttts PTAL

zjj2wry commented:
/lgtm

WanLinghao commented:
@zjj2wry do you know who can help me push this patch, thanks a lot

WanLinghao commented:
@BenTheElder do you know who can help me push this,  thanks a lot

fabianofranz commented:
@WanLinghao thanks!
/approve no-issue
/sig cli
